lib/sign.js

```diff
@@ -17,11 +18,51 @@ var xpi = require("./xpi");
 
 var AMO_API_PREFIX = require("./settings").AMO_API_PREFIX;
 
+function getXpiInfo(options) {
+  options = _.assign({
+    addonDir: null,
+    xpiPath: null,
+  }, options);
+
+  if (!options.xpiPath) {
+    return utils.getManifest({addonDir: options.addonDir})
+      .then(function(manifest) {
+        return getXpiInfoFromManifest(manifest);
+      });
+  } else {
+    return utils.extractXPI(options.xpiPath)
```

Comment:
Ah, thanks, I took a closer look and I think I can merge the two. It seems that `sign` is the only one that passes `xpiPath` to `getManifest()` so I can change its behavior. Before, I was scared to change how it handled missing manifest files :fearful: